src/plugins/monitoring/src/java/com/reucon/openfire/plugin/archive/impl/JdbcPersistenceManager.java

```diff
@@ -456,7 +456,7 @@ private int bindConversationParameters(Date startDate, Date endDate, String owne
 			appendWhere(whereSB, "( ", MESSAGE_TO_JID, " = ? OR ", MESSAGE_FROM_JID, " = ? )");
 		}
 		if (whereSB.length() != 0) {
-			querySB.append(" AND ").append(whereSB);
+			querySB.append(" WHERE ").append(whereSB);
 		}
```

Comment:
Not quite right - SELECT_MESSAGES_ORACLE has no WHERE clause, but SELECT_MESSAGES always does. I think the right fix is therefore to add a dummy WHERE clause to the Oracle statement (WHERE 1=1 or something).

Reply:
Thanks, I agree!